spark/v3.4/spark-extensions/src/main/scala/org/apache/iceberg/spark/extensions/IcebergSparkSessionExtensions.scala

```diff
@@ -51,13 +51,13 @@ class IcebergSparkSessionExtensions extends (SparkSessionExtensions => Unit) {
     extensions.injectResolutionRule { _ => ProcedureArgumentCoercion }
     extensions.injectResolutionRule { _ => AlignRowLevelCommandAssignments }
     extensions.injectResolutionRule { _ => RewriteUpdateTable }
-    extensions.injectResolutionRule { _ => RewriteMergeIntoTable }
     extensions.injectCheckRule { _ => MergeIntoIcebergTableResolutionCheck }
     extensions.injectCheckRule { _ => AlignedRowLevelIcebergCommandCheck }
 
     // optimizer extensions
     extensions.injectOptimizerRule { _ => ExtendedSimplifyConditionalsInPredicate }
     extensions.injectOptimizerRule { _ => ExtendedReplaceNullWithFalseInPredicate }
+    extensions.injectOptimizerRule { spark => RewriteStaticInvoke(spark) }
```

Comment:
Put in the optimizer in order to some constant expressions have been optimized. Such as `system.days(ts) = system.days(date('2022-09-08'))`. The `system.days(date('2022-09-08')` will be evaluated into a constant value.

Reply:
Can we do this in a separate PR? I don't think the basic support for functions should include this.

Reply:
I intend to do that, however, SOL uts depends on it. I will submit a separate PR to do this.

Reply:
Submit #8088 and will add the sql push-down UTs after this in.